src/pages/BlogPage.js

```diff
@@ -0,0 +1,77 @@
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import axiosInstance from "../api/axios";
+import "./pageStyle/BlogPage.css";
+
+const BlogPage = () => {
+  const [displayBlogs, setDisplayBlogs] = useState([]);
+  const [page, setPage] = useState(1);
+  let navigate = useNavigate()
+  const getBlog = async () => {
+    try {
+      const { data } = await axiosInstance.get();
+      const allPost = data.getBlogPost.docs
+      setDisplayBlogs((prevPost) => [...prevPost, ...allPost]);
+      loadMore()
+    } catch {
+      console.log("Error occured");
+    }
+  };
+
+  const loadMore = () => {
+    setPage(page + 1);
+  };
+
+  const handleScroll = (e) => {
+    if(window.innerHeight + e.target.documentElement.scrollTop + 1 >=
+      e.target.documentElement.scrollHeight) {
+        getBlog()
+      }
+  }
+
+  useEffect(() => {
+    getBlog();
+    window.addEventListener('scroll', handleScroll)
+  },[]);
```

Comment:
`getBlog` definition should be moved inside this `useEffect`

There should be a cleanup function for the eventListener

Reply:
ok